### ISSUE TITLE
is v2 up to date?

### ISSUE THREAD

**Issue by nknapp** [kriskowal/q-io#156]
I've noticed that there have been many commits in v1 and v2 without any merging. Issues like #149 have only been resolved in v1. Is the v2-branch abandoned? Would it make sense to merge v1 into v2 to apply the remaining changes? Or are the differences so large that they cannot be merged?

I want to use v2 now, because of montagejs/collections#139

**Comment by dogancelik** [comment]
Please merge

**Comment by kriskowal** [comment]
A straight merge is not possible. v2 is substantially different. These patches need to be manually reproduced on v2 to land. PR’s are welcome. Sorry for the delay—I will try to grow the contributor base.

**Comment by nknapp** [comment]
I attempted a merge but there were many conflicts and I had the feeling that I was not able to resolve the conflicts as it would have been expected.

I worked around the problem by building an alternative library (`m-io`) with a reimplementation of the few functions that I need for my project, with no dependency to the `collections` library. 

I don't like the idea of publishing duplicates to npm, but in this case (and the frustration of a morning spent tracing a bug caused by montagejs/collections#139) it was the solution with the least effort.
